### ISSUE TITLE
Bonsai 's relay REST /v1/callbacks accepts only bincode serialized data

### ISSUE THREAD

**Issue by DefiCake** [risc0/risc0#939]
This limits the possibility of crafting CallBackRequests in other languages and limits it to rust. 

```rs
/// Post a new [CallbackRequest] to Bonsai.
    pub async fn callback_request(
        &self,
        request: impl Into<CallbackRequest>,
    ) -> Result<(), ClientError> {
        let res = self
            .client
            .post(format!("{}{CALLBACK_ROUTE}", self.url))
            .header(header::CONTENT_TYPE, "application/octet-stream")
            .body(bincode::serialize(&request.into())?)
            .send()
            .await?;
        error_for_status(res).await?;

        Ok(())
    }
```

A more widely accepted format would be JSON bodies or BASE64 parameters

**Comment by lazyfuhrer** [comment]
Can I work on this ? @DefiCake

**Comment by DefiCake** [comment]
> Can I work on this ? @DefiCake

I am not part of the team. I opened a PR, feel free to fork it and improve it, I am still very green in rust so I don't think it will be merged.

A thing I have noticed while playing with this, is that the bytes arrays `image_id` and `input` must be truly numbers. If you attempt to use an `Uint8Array` from node and send a post request, these numbers will be encoded as strings (think `input: ["0", "17", "243"...]` vs `input: [0, 17, 243...]`) and the deserialization will fail.
